fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/bitmap/ComposedBitmapIndexAggregate.java

```diff
@@ -99,12 +100,18 @@
      * @return an {@code Optional} query plan or {@code Optional.empty} if planning is not possible
      */
     @Nonnull
-    public Optional<ComposedBitmapIndexQueryPlan> tryPlan(@Nonnull RecordQueryPlanner planner,
-                                                          @Nonnull RecordQuery.Builder queryBuilder) {
+    public Optional<RecordQueryPlan> tryPlan(@Nonnull RecordQueryPlanner planner,
+                                             @Nonnull RecordQuery.Builder queryBuilder) {
         final List<RecordQueryCoveringIndexPlan> indexScans = new ArrayList<>();
         final Map<IndexNode, ComposedBitmapIndexQueryPlan.IndexComposer> indexComposers = new IdentityHashMap<>();
-        return Optional.ofNullable(plan(root, queryBuilder, planner, indexScans, indexComposers))
-                .map(composer -> new ComposedBitmapIndexQueryPlan(indexScans, composer));
+        final ComposedBitmapIndexQueryPlan.ComposerBase composer = plan(root, queryBuilder, planner, indexScans, indexComposers);
+        if (composer == null || indexScans.isEmpty()) {
+            return Optional.empty();
+        }
+        if (indexScans.size() == 1) {
+            return Optional.ofNullable(indexScans.get(0));
```

Comment:
Is this correct if there's a `NOT` in the composer tree against a single index scan?

Also, separately, without this change, would it have returned a `ComposedBitmapIndexQueryPlan` with a single `IndexNode` in the tree (which I think is correct, but perhaps inefficient), or would it have returned something less good?

Reply:
It would have given an error, insisting that there is more than one child.

Reply:
I added a check to make sure the composer is trivial (pass through the index). It _might_ be possible to get `NOT` to work in this case, but it doesn't look like it would today, even if the composer were given a chance.

Reply:
Okay, sounds good. I suppose it would be nice if `NOT` planned correctly, but having it fail to plan is fine (but having it return the wrong plan would have seemed dangerous).